Tests/Tests/Core/Extensions/UIKit/UIViewTests.swift

```diff
@@ -0,0 +1,133 @@
+  private let shadowColor: UIColor = .red
+  private let shadowRadius: CGFloat = 50
+  private let shadowOpacity: Float = 0.5
+  private let shadowOffset: CGSize = .init(width: 5, height: 10)
+  
+  func test_dropShadows_allParameters() {
+    let view = UIView()
+    view.layer.shadowColor = UIColor.blue.cgColor
+    XCTAssertNil(view.layer.shadowPath)
+    XCTAssertEqual(view.layer.shadowColor, UIColor.blue.cgColor)
+    XCTAssertEqual(view.layer.shadowOffset, .init(width: 0, height: -3))
+    XCTAssertEqual(view.layer.shadowOpacity, 0)
+    XCTAssertEqual(view.layer.shadowRadius, 3)
+    
+    view.dropShadow(path: shadowPath, shadowColor: shadowColor, radius: shadowRadius, opacity: shadowOpacity, offset: shadowOffset)
+    XCTAssertEqual(view.layer.shadowPath, shadowPath.cgPath)
+    XCTAssertEqual(view.layer.shadowColor, shadowColor.cgColor)
+    XCTAssertEqual(view.layer.shadowOffset, shadowOffset)
+    XCTAssertEqual(view.layer.shadowOpacity, shadowOpacity)
+    XCTAssertEqual(view.layer.shadowRadius, shadowRadius)
+  }
+  
+  func test_dropShadows_missingPath() {
+    let view = UIView()
+    XCTAssertNil(view.layer.shadowPath)
+    
+    view.dropShadow(path: nil, shadowColor: shadowColor, radius: shadowRadius, opacity: shadowOpacity, offset: shadowOffset)
+    XCTAssertNil(view.layer.shadowPath)
+  }
+  
+  func test_applyBorder() {
+    let color: UIColor = .blue
+    let width: CGFloat = 10
+    
+    let view = UIView()
+    view.layer.borderColor = UIColor.black.cgColor
+    XCTAssertEqual(view.layer.borderColor, UIColor.black.cgColor)
+    XCTAssertEqual(view.layer.borderWidth, 0)
+    
+    view.applyBorder(color: color, width: width)
+    XCTAssertEqual(view.layer.borderColor, color.cgColor)
+    XCTAssertEqual(view.layer.borderWidth, width)
+  }
+  
+  func test_setHidden_true() {
+    let view = UIView()
+    
+    XCTAssertFalse(view.isHidden)
+    XCTAssertEqual(view.alpha, 1)
+    
+    let promise = expectation(description: "Wait for animation...")
+    view.setHidden(true, animationDuration: 0.1) {
+      promise.fulfill()
+    }
+    
+    waitForExpectations(timeout: 0.5)
+    
+    XCTAssertTrue(view.isHidden)
+    XCTAssertEqual(view.alpha, 0)
+  }
+  
+  func test_setHidden_false() {
+    let view = UIView()
+    view.isHidden = true
+    view.alpha = 0
+    
+    let promise = expectation(description: "Wait for animation...")
+    view.setHidden(false, animationDuration: 0.1) {
+      promise.fulfill()
+    }
+    
+    waitForExpectations(timeout: 0.5)
+    
+    XCTAssertFalse(view.isHidden)
+    XCTAssertEqual(view.alpha, 1)
+  }
+  
+  func test_rotate_clockwise() {
+    let view = UIView()
+    
+    view.rotate(speed: 1, clockwise: true)
+    let animation = view.layer.animation(forKey: UIView.AnimationKey.rotation) as? CABasicAnimation
+    XCTAssertNotNil(animation)
+    XCTAssertEqual(animation?.fromValue as? Int, 0)
+    XCTAssertEqual(animation?.toValue as? Float, (Float.pi * 2.0) * 1)
```

Comment:
We should avoid magic numbers.

Reply:
There is no magic in this number, pure math :)

Reply:
> There is no magic in this number, pure math :)

Do you not agree that it would be nicer to declare them in a property, that way we know immediately what they represent?

`let expectedStartAngle = 0`
`let expectedEndAngle = (Float.pi * 2.0) * 1`

Reply:
That would be a nice addition yes, but not necessary.